src/Groups/MatGrp.jl

```diff
@@ -0,0 +1,892 @@
+      vero || throw(ArgumentError("Element not in the group"))
+      if x_gap==nothing return MatrixGroupElem(G,x.elm)
+      else return MatrixGroupElem(G,x.elm,x_gap)
+      end
+   end
+end
+
+# embedding a nxn array into a group G
+function (G::MatrixGroup)(L::AbstractVecOrMat; check=true)
+   x = matrix(G.ring, G.deg, G.deg, L)
+   return G(x; check=check)
+end
+
+
+########################################################################
+#
+# Methods on elements
+#
+########################################################################
+
+# we are not currently keeping track of the parent; two elements coincide iff their matrices coincide
+function ==(x::MatrixGroupElem{S,T},y::MatrixGroupElem{S,T}) where {S,T}
+   if isdefined(x,:X) && isdefined(y,:X) return x.X==y.X
+   else return x.elm==y.elm
+   end
+end
+
+# Base.:* is defined in src/Groups/GAPGroups.jl, and it calls the function _prod below
+# if the parents are different, the parent of the output product is set as GL(n,q)
+function _prod(x::MatrixGroupElem,y::MatrixGroupElem)
+   G = x.parent==y.parent ? x.parent : GL(x.parent.deg, x.parent.ring)
+   if isdefined(x,:X) && isdefined(y,:X) && !(isdefined(x,:elm) && isdefined(y,:elm))
+      return MatrixGroupElem(G, x.X*y.X)
+   else
+      return MatrixGroupElem(G, x.elm*y.elm)
+   end
+end
+
+Base.:*(x::MatrixGroupElem, y::fq_nmod_mat) = x.elm*y
+Base.:*(x::fq_nmod_mat, y::MatrixGroupElem) = x*y.elm
+
+Base.:^(x::MatrixGroupElem, n::Int) = MatrixGroupElem(x.parent, x.elm^n)
+
+Base.isone(x::MatrixGroupElem) = isone(x.elm)
+
+Base.inv(x::MatrixGroupElem) = MatrixGroupElem(x.parent, inv(x.elm))
+
+# if the parents are different, the parent of the output is set as GL(n,q)
+function Base.:^(x::MatrixGroupElem, y::MatrixGroupElem)
+   G = x.parent==y.parent ? x.parent : GL(x.parent.deg, x.parent.ring)
+   if isdefined(x,:X) && isdefined(y,:X) && !(isdefined(x,:elm) && isdefined(y,:elm))
+      return MatrixGroupElem(G, inv(y.X)*x.X*y.X)
+   else
+      return MatrixGroupElem(G,inv(y.elm)*x.elm*y.elm)
+   end
+end
+
+comm(x::MatrixGroupElem, y::MatrixGroupElem) = inv(x)*conj(x,y)
+
+"""
+    det(x::MatrixGroupElem)
+    
+Return the determinant of `x`.
+"""
+det(x::MatrixGroupElem) = det(x.elm)
+
+"""
+    base_ring(x::MatrixGroupElem)
+
+Return the base ring of `x`.
+"""
+base_ring(x::MatrixGroupElem) = x.parent.ring
+
+parent(x::MatrixGroupElem) = x.parent
+
+"""
+    matrix(x::MatrixGroupElem)
+
+Return the underlying `AbstractAlgebra` matrix of `x`.
+"""
+matrix(x::MatrixGroupElem) = x.elm
+
+Base.getindex(x::MatrixGroupElem, i::Int, j::Int) = x.elm[i,j]
+
+"""
+    nrows(x::MatrixGroupElem)
+
+Return the number of rows of the given matrix.
+"""
+nrows(x::MatrixGroupElem) = x.parent.deg
+
+"""
+    trace(x::MatrixGroupElem)
+    tr(x::MatrixGroupElem)
+
+Return the trace of `x`.
+"""
+trace(x::MatrixGroupElem) = trace(x.elm)
+tr(x::MatrixGroupElem) = tr(x.elm)
+
+order(x::MatrixGroupElem) = GAP.Globals.Order(x.X)
```

Comment:
I commented on this in another PR, suggesting to turn this into
```suggestion
order(x::MatrixGroupElem) = ZZ(GAP.Globals.Order(x.X))
```
but actually to match the generic `order` for GAP groups, we probably should also have this:
```suggestion
order(x::MatrixGroupElem) = ZZ(GAP.Globals.Order(x.X))
order(::Type{T}, x:: MatrixGroupElem) = T(GAP.Globals.Order(x.X))
```